src/Voting.js

```diff
@@ -0,0 +1,96 @@
+import {useState, useEffect} from 'react';
+import firebase from './firebase';
+import {getDatabase, ref, update, get, increment} from 'firebase/database';
+import uuid from 'react-uuid';
+
+export default function Voting({code, userHasVoted, setUserHasVoted}) {
+
+  const [pollQuestion, setPollQuestion] = useState(``);
+  const [choices, setChoices] = useState([]);
+  const [userChoice, setUserChoice] = useState('');
+
+
+  // Generate the poll options
+  useEffect(() => {
+    const database = getDatabase(firebase);
+    const dbRef = ref(database, code);
+    
+    get(dbRef).then((snapshot) => {
+      if (snapshot.exists()) {
+        const data = snapshot.val()
+        let newQuestion = data.question;
+        if (newQuestion[newQuestion.length -1] !== '?') {
+          newQuestion += '?';
+        }
+        setPollQuestion(newQuestion);
+        const newChoices = [];
+        for (let item in data.options) {
+          newChoices.push(item);
+        }
+        setChoices(newChoices);
+      } else {
+        setPollQuestion(`We couldn't find your poll.`)
+      }
+    })
+  }, [code]);
+
+  function handleChange(e) {
+    setUserChoice(e.target.value);
+  }
+
+  function handleKeyDown(e) {
+    if (e.key === 'Enter' || e.key === ' ') {
+      setUserChoice(e.target.innerText);
+    }
+  }
+
+  // Submit the user's vote and note that they have voted in both state and localStorage so that <DisplayPoll /> displays <Results /> instead.
+  function handleSubmit(e) {
+    e.preventDefault();
+    if (userHasVoted) {
+      alert('You have already voted in this poll.')
```

Comment:
feel like prompt might be the better option

Reply:
Wouldn't prompt just ask them to provide text? I should mention that it should be impossible for anyone to ever see this text, as the Voting component should never render if the user has already voted.